### PR TITLE
Fix add new Budget form

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
- add validation for required fields
- fix creating a budget name
- fix budger status options

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Budget/constants.js [hunk 1]
```diff
@@ -8,12 +8,7 @@ export const BUDGET_STATUSES = {
   PLANNED: 'Planned',
 };
 
-export const BUDGET_STATUSES_OPTIONS = [
-  { label: 'ui-finance.budget.status.active', value: BUDGET_STATUSES.ACTIVE },
-  { label: 'ui-finance.budget.status.closed', value: BUDGET_STATUSES.CLOSED },
-  { label: 'ui-finance.budget.status.frozen', value: BUDGET_STATUSES.FROZEN },
-  { label: 'ui-finance.budget.status.planned', value: BUDGET_STATUSES.PLANNED },
-];
+export const BUDGET_STATUSES_OPTIONS = Object.values(BUDGET_STATUSES).map(status => ({ label: status, value: status }));
```
Comment (aliaksei-chumakou) [inline src/components/Budget/constants.js]: but label shouldn't be the same as value. It should be translated. There is only way - to use `labelId`. So I suppose the proper fix would be to change `label` to `labelId` above

File: src/components/Budget/AddBudgetModal/AddBudgetModal.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ const AddBudgetModal = ({ history, mutator, resources, onClose, fund, budgetStat
         const budget = await mutator.budget.POST({
           ...formValue,
           fundId: fund.id,
-          name: `${fund.code}-${fiscalYear.code}`,
+          name: `${fund.code}-${fiscalYear.label}`,
```
Comment (aliaksei-chumakou) [inline src/components/Budget/AddBudgetModal/AddBudgetModal.js]: so fiscalYear doesn't have `code` field anymore?

File: src/components/Budget/constants.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/components/Budget/constants.js]: Right. I didn't update my local `node_modules` and used non-actual code from `stripes-acq-components`

File: src/components/Budget/AddBudgetModal/AddBudgetModal.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/components/Budget/AddBudgetModal/AddBudgetModal.js]: It has, but here `fiscalYear.code` = `fiscalYear.label`. `fiscalYear.code` is required. See https://github.com/folio-org/ui-finance/pull/156/files#diff-4337fea6055306941a9b97d7f6896821R21 and https://github.com/folio-org/ui-finance/blob/master/src/Utils/getFiscalYearsForSelect.js#L5

**Pull request merged**